viewing_party/party.py

```diff
@@ -1,23 +1,173 @@
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        movie_dict = {}
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+    else:
+        return None
+    
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+
+#loop through the dictionaries in the list user_data["watchlist"]
+#if the dictionary key == "title", check if the value == title (parameter)
+#if the value == title, save the dictionary to user_data["watched"]. Then delete the dictionary from user_data["watchlist"]
+
+
+    for dict in user_data["watchlist"]:
+        if dict["title"] == title:
+            user_data["watched"].append(dict)
+            user_data["watchlist"].remove(dict)
+
+    return user_data
+
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
+
+def get_watched_avg_rating(user_data):
+
+    ratings_list = []
+    
+    if len(user_data["watched"]) > 0:
+        for dict in user_data["watched"]:
+            ratings_list.append(dict["rating"])
+        
+        average_rating = sum(ratings_list) / len(ratings_list)
+        
+    else:
+        average_rating = 0.0
+    
+    return average_rating
+    
+    
+def get_most_watched_genre(user_data):
+
+    genre_frequency = {}
+
+
+    if len(user_data["watched"]) > 0:
+        for dict in user_data["watched"]:
+            for k, v, in dict.items():
+                if k == "genre":
+                    if v not in genre_frequency:
+                        genre_frequency[v] = 1
+                    else:
+                        genre_frequency[v] += 1
+
+ 
+        most_frequent_genre = max(genre_frequency, key=genre_frequency.get)
+    
+    else:
+        most_frequent_genre = None
+
+    return most_frequent_genre
 
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
 
+def get_unique_watched(user_data):
+
+    friend_watched = []
+    difference = []
+
+    for dict in user_data["friends"]:
+        for k, v, in dict.items():
+                if k == "watched":
+                    friend_watched.extend(v)
+        
+    for dict in user_data["watched"]:
+        if dict not in friend_watched:
+            difference.append(dict)
+
+    return difference
+
+
+def get_friends_unique_watched(user_data):
+
+    friend_watched = []
+    difference1 = []
+    difference2 = []
+
+    for dict in user_data["friends"]:
+        for k, v, in dict.items():
+                if k == "watched":
+                    friend_watched.extend(v)

```

Comment:
Since we make a list of all our friends' movies in this function & the one above it, the code to do that is a prime candidate to move to a helper function where we can reuse it!